Revised TechDoc Analysis Tools (#229)
cncf/techdocs: TechDoc-Assistance-Program.md

Reviewer: Rename to `techdocs-assistance.md`.

Reviewer 2: I agree, and think I prefer just `assistance.md` here.

> # Technical Documentation Assistance Program

Reviewer: ```suggestion
# Assistance program for technical documentation
```

This document outlines the Cloud Native Computing Foundation (CNCF) Technical Documentation Assistance Program (the Program), a service offered by CNCF Tech Docs for evaluating and improving an OSS project's technical documentation. The process is designed to:

> 1. Provide a baseline analysis of the project's documentation quality measured against the project's [maturity level](analysis/analysis-tools/criteria.md. (Often, documentation help is requested when a project requests promotion to a new maturity level.)

Reviewer: ```suggestion
1. Provide a baseline analysis of the project's documentation quality measured against the project's [maturity level](analysis/analysis-tools/criteria.md). Often, projects request an analysis in support of promotion to a new maturity level.
```

1. Recommend changes that will reduce the gap between the documentation and the maturity of the overall project.
1. Expand on the recommended changes in an implementation plan.
1. Break down the implementation into a documentation project backlog comprising a GitHub Issues list.
1. Support the project team's contributors in taking up and completing the issue backlog items.
1. Leave the project team with an improved understanding and skill base for improving and maintaining the project documentation.

## Phase 0: Training

Some level of familiarity with the technical documentation process is required to:

- Work effectively with technical writers
- Draft technical documentation (for non-writers)

> - Improve the chances of a good outcome from the Tech Doc Assistance Program

Reviewer: ```suggestion
- Get the best results out of the Assistance Program
```

> For this reason, CNCF offers free training on documentation essentials for project contributors and maintainers. To get the most from the Technical Documentation Assistance Program, project contributors should do the training *before* engaging a documentation specialist to complete the documentation analysis.

Reviewer: ```suggestion
For this reason, CNCF offers free training on documentation essentials for project contributors and maintainers. To get the most from the Assistance Program, project contributors are encouraged to do the training *before* engaging a documentation specialist to complete the documentation analysis.
```

Should the last part be simplified to this:

```suggestion
For this reason, CNCF offers free training on documentation essentials for project contributors and maintainers. To get the most from the Assistance Program, project contributors should do the training *before* applying for the program.
```

Author: That seems like a good edit.

The training program consists of the following online courses. Anyone can sign up for and complete the courses at their own pace.

1. [Open Source Technical Documentation Essentials (LFC111)](https://training.linuxfoundation.org/training/open-source-technical-documentation-essentials-lfc111/)
1. [Creating Effective Documentation for Developers (LFC112)](https://training.linuxfoundation.org/training/creating-effective-documentation-for-developers-lfc112/)

> ## Phase 1: Documentation Analysis

Reviewer: ```suggestion
## Phase 1: Documentation analysis
```

> A technical writer (on CNCF staff or on contract) analyzes the documentation. Based on the standards developed by the CNCF Tech Docs program, the writer:

Reviewer: ```suggestion
A technical writer (on CNCF staff or on contract) analyzes the documentation. Based on the standards developed as part of the CNCF TechDocs program, the writer:
```

> 1. Estimates the maturity level of the documentation compared to the current or desired maturity level of the software project, based on a number of criteria in three categories, using a rubric developed by CNCF. The categories are:

Reviewer: A bit of a run-on sentence, consider revising.

> 1. Estimates the maturity level of the documentation compared to the current or desired maturity level of the software project, based on a number of criteria in three categories, using a rubric developed by CNCF. The categories are:

Author: ```suggestion
1. Estimates the maturity level of the documentation compared to the current or desired maturity level of the software project using a rubric developed by CNCF. The rubric is divided into three categories:
2. ```

    1. Project documentation: The end-user documentation for the project's work product, typically (but not always) an application, API, protocol, or some other software product.
    1. Contributor documentation: Documentation about the project, aimed at project contributors and describing procedures, infrastructure, and customs for doing project work. This includes artifacts that define procedures and governance; recruit, orient, and train project contributors; and name responsible parties (project leaders, often generically called *maintainers*).
    1. Website: The technical infrastructure behind the documentation and the project's web presence, including website generation, versioning, SEO, analytics, and security.
1. Collaborates with project leadership to identify user roles and objectives for software users. 

> 1. Proposes changes, if necessary, to the organization and content of the documentation to close gaps with the proposed maturity level.

Reviewer: ```suggestion
1. Proposes changes, if necessary, to the organization and content of the documentation to close gaps with the target maturity level.
```

> 1. Writes an implementation plan describing improvements that address the gaps identified in the analysis.

Reviewer: ```suggestion
1. Writes an implementation plan describing improvements that address the gaps identified by the analysis.
```

> ## Phase 2: Backlog Creation

Reviewer: ```suggestion
## Phase 2: Backlog creation
```

Once a high-level improvement plan has been written and approved, the tech writer analyzes the proposed changes to create a backlog of actionable, individual writing assignments and other tasks to improve the documentation. These tasks should have two primary characteristics:
- As much as possible, they should be independent of each other so they can be completed in any order by any combination of contributors; and

> - They should be time-constrained. If possible, large tasks should be broken down into smaller, independent tasks that require hours, not days, to complete.

Reviewer: I'd say days can be ok, but not weeks, maybe something like:

```suggestion
- They should be time-constrained. If possible, large tasks should be broken down into smaller, independent tasks that require hours or days rather than weeks to complete.
```

> ## Phase 3: Documentation Improvement

Reviewer: ```suggestion
## Phase 3: Documentation improvement
```

Community members work on the issues created in the previous phase. Ideally, tech writers are available to advise contributors and edit work, especially if the contributing community members are not trained technical writers. Remember that the training courses in Phase 0 are available to prepare contributors with general knowledge of the process.

> We know that recruiting contributors to write documentation can be difficult. While this is largely the responsibility of the project leadership, CNCF is actively working on ways to encourage doc contributions. For example, creating a backlog of time-bounded issues is an attempt to lower barriers, both psychological and logistical, to documentation creation and maintenance. 

Reviewer: @nate-double-u - Do we want to mention mentorship opportunities twice(?) a year?

> ## Phase 4: Impact Analysis

Reviewer: ```suggestion
## Phase 4: Impact analysis
```

> Measurement is vital to any process improvement effort, and this is currently the weakest point of the TechDoc Assistance Program. We do no offer any tools to measure the impact of doc improvements, although we can assure you that they are profound. Projects are encouraged to collect metrics on documentation effectiveness. 

Reviewer: ```suggestion
Projects are encouraged to collect metrics (using Google analytics and page feedback data) on documentation usage as a means of assessing the effectiveness of documentation improvements.
```

> As a basic gauge the effectiveness of the documentation effort, projects can track the issue backlog. This can provide a guide to progress on leveling the documentation maturity to that of the project.

Reviewer: Hmm, I'm not sure about this paragraph. How about (or just drop the paragraph IMHO):

```suggestion
Projects can also track the issue backlog as a means of measuring the increased documentation maturity.
```

Author: Agreed.
